File: src/main/java/com/hjj/apiserver/controller/MainController.java
package com.hjj.apiserver.controller;

import com.hjj.apiserver.common.ApiError_Java;
import com.hjj.apiserver.common.ApiResponse_Java;
import com.hjj.apiserver.common.ApiUtils;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController
@RequiredArgsConstructor
public class MainController {

    private final MainService mainService;

    @GetMapping("/main")
    public ApiResponse_Java main(){
        try{
            return ApiUtils.success(mainService.findMain());
        } catch (Exception e) {
            log.error("main Error exception: {}", e);
            return ApiUtils.error(ApiError_Java.ErrCode.ERR_CODE0005.getMsg(), ApiError_Java.ErrCode.ERR_CODE0005);
        }
    }
}

File: src/main/kotlin/com/hjj/apiserver/controller/MainController.kt
package com.hjj.apiserver.controller

import com.hjj.apiserver.dto.main.response.MainFindResponse
import com.hjj.apiserver.service.impl.MainService
import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@Api(tags = ["8. Main"])
@RestController
class MainController(
    private val mainService: MainService,
) {
    @ApiOperation(value = "메인 화면 조회", notes = "메인에서 보여줄 데이터를 조회한다.")
    @GetMapping("/main")
    fun main(): MainFindResponse {
        return mainService.findMain()
    }
}
